openmc/mgxs/groups.py

```diff
@@ -0,0 +1,275 @@
+        self._group_edges = np.array(edges)
+        self._num_groups = len(edges)-1
+
+    def generate_bin_edges(self, start, stop, num_groups, spacing='linear'):
+        """Generate equally or logarithmically-spaced energy group boundaries.
+
+        Parameters
+        ----------
+        start : Real
+            The lowest energy in MeV
+        stop : Real
+            The highest energy in MeV
+        num_groups : Integral
+            The number of energy groups
+        spacing : {'linear', 'logarithmic'}
+            The spacing between groups
+
+        """
+
+        cv.check_type('first edge', start, Real)
+        cv.check_type('last edge', stop, Real)
+        cv.check_type('number of groups', num_groups, Integral)
+        cv.check_type('spacing', spacing, basestring)
+        cv.check_greater_than('first edge', start, 0, True)
+        cv.check_greater_than('last edge', stop, start, False)
+        cv.check_greater_than('number of groups', num_groups, 0)
+        cv.check_value('spacing', spacing, ('linear', 'logarithmic'))
+
+        if spacing == 'linear':
+            self.group_edges = np.linspace(start, stop, num_groups + 1)
+        elif spacing == 'logarithmic':
+            self.group_edges = \
+                np.logspace(np.log10(start), np.log10(stop), num_groups + 1)
+
+        self._num_groups = num_groups
+
+    def get_group(self, energy):
+        """Returns the energy group in which the given energy resides.
+
+        Parameters
+        ----------
+        energy : Real
+            The energy of interest in MeV
+
+        Returns
+        -------
+        Integral
+            The energy group index, starting at 1 for the highest energies
+
+        Raises
+        ------
+        ValueError
+            If the group edges have not yet been set.
+
+        """
+
+        if self.group_edges is None:
+            msg = 'Unable to get energy group for energy "{0}" MeV since ' \
+                  'the group edges have not yet been set'.format(energy)
+            raise ValueError(msg)
+
+        index = np.where(self.group_edges > energy)[0]
+        group = self.num_groups - index
+        return group
+
+    def get_group_bounds(self, group):
+        """Returns the energy boundaries for the energy group of interest.
+
+        Parameters
+        ----------
+        group : Integral
+            The energy group index, starting at 1 for the highest energies
+
+        Returns
+        -------
+        2-tuple
+            The low and high energy bounds for the group in MeV
+
+        Raises
+        ------
+        ValueError
+            If the group edges have not yet been set.
+
+        """
+
+        if self.group_edges is None:
+            msg = 'Unable to get energy group bounds for group "{0}" since ' \
+                  'the group edges have not yet been set'.format(group)
+            raise ValueError(msg)
+
+        lower = self.group_edges[self.num_groups-group]
```

Comment:
You should check the value of group passed. Otherwise, if you pass group > self.num_groups, it works and will use a negative index, probably not an intended effect.